metadata-ingestion/src/datahub/ingestion/source/mlflow.py

```diff
@@ -262,6 +273,186 @@ def _convert_run_result_type(
                 type="SKIPPED", nativeResultType=self.platform
             )
 
+    def _get_dataset_schema(self, schema: str) -> Optional[List[Tuple[str, str]]]:
+        try:
+            schema_dict = json.loads(schema)
+        except json.JSONDecodeError:
+            print("Failed to parse schema JSON")
```

Comment:
no prints - use logger instead

also maybe include the invalid json in the message?